pkg/daemon/daemon.go

```diff
@@ -323,6 +320,9 @@ func (dn *Daemon) ClusterConnect(
 	dn.mcLister = mcInformer.Lister()
 	dn.mcListerSynced = mcInformer.Informer().HasSynced
 
+	dn.nodeWriter = newNodeWriter(dn.name, dn.kubeClient.CoreV1().Nodes(), dn.nodeLister)
```

Comment:
Do you plan on creating both kubeclients in this main connect function, and then passing one to the nodewriter? I think having nodewriter maintaining it's own kubeclient (as you did in https://github.com/openshift/machine-config-operator/pull/3141) could work, and then the MCD SA could just have read/list/get to read off of the node. (Which would match the kubelet's perms according to https://kubernetes.io/docs/reference/access-authn-authz/node/ so its no extra harm)

Alternatively, we could have dn.kubeClient be initialized off of kubelet and have both nodewriter and regular sync loop use that, and then have just a special client for reading MC objects used by the update loop. Which do we prefer?